src/preferences/upgrade.cpp

```diff
@@ -433,9 +433,42 @@ UserSettingsPointer Upgrade::versionUpgrade(const QString& settingsPath) {
         }
     }
 
-    if (configVersion.startsWith("1.12") ||
-        configVersion.startsWith("2.0") ||
-        configVersion.startsWith("2.1.0")) {
+    if (configVersion.startsWith("1.12")) {
```

Comment:
The version check and upgrade should be implemented generically, i.e. not requiring changes for each new version. Only exceptions from the general rule must be handled separately. Currently, there are no exceptions. Anything else is not maintainable and a burden.

Reply:
This PR actually does the the maintenance to avoid the warning:
`Warning [Main]: Configuration file is at version "2.2.1" instead of the current 2.2.2`
A burden that has been introduced in an early version of Mixxx.  
So we my consider to merge this solution just as a band aid following he legacy pattern.   

However I agree to @uklotzde comment that a generic solution would be much better and also only requires  a few lines of code. 
How about introducing dropping the requirement that the config version matches the Mixxx version. 
We may introduce a "MinConfigVersion" or such and check if the current version is >= the min version.
  
What do you think?

Reply:
> So we my consider to merge this solution just as a band aid following he legacy pattern.

If we need a bandaid to suppress the logged warning, then let's at least combine all upgrades 1.11++ that don't require cfg key migration (and not do incremental upgrades for each major version, like when going from 1.11 to 2.3)

Reply:
Hello, I apologize for the late reply.
I went with this approach assuming having this incremental approach was important to the upgrade process.

If I understand right, are we trying to just look at a way to check if the config is at the current version or not?

I think I can implement the minConfigVersion recommended by @daschuer, though I was wondering how often that minConfigVersion would have to be updated? Is there a particular mismatch in config files between different versions? Unfortunately i hadn't gone into the background of the issue.

@uklotzde had mentioned a solution that used regex, would you please mind elaborating what functionality wed like to achieve, is it just a check for the main release version?

Reply:
The solutions needs to consider both upgrades and downgrades, i.e. opening a newer config file with an older version. Unfortunately, I don't have any concrete ideas on how to achieve this. Proposing a solution is part of the work for fixing this bug.

Reply:
I don think we need to downgrade the preferences. If one tests an older version he will loose the the settings stored in the new version. Since now we also do no downgrade. This applies also for the database schema by the way. Instead we store a MinBackwardsCompatibleVersion in the database. 
https://github.com/mixxxdj/mixxx/blob/e405c4770ee105b40555a711a3e224fb9305f5d5/src/database/schemamanager.cpp#L123

> though I was wondering how often that minConfigVersion would have to be updated? Is there a particular mismatch in config files between different versions? Unfortunately i hadn't gone into the background of the issue.

You see the whole history in the upgrade.cpp cpp file. The future is hard to predict, at some point we will migrate away from the single .mixxx folder: https://bugs.launchpad.net/mixxx/+bug/1463273